spec/ysh-regex-api.test.sh

```diff
@@ -84,6 +84,44 @@ yes
 yes
 ## END
 
+#### Eggex flags to treat newlines as special are respected
+shopt -s ysh:upgrade
+
+if ($'abc123\n' ~ / digit $ /) {
+  echo 'BUG'
+}
+if ($'abc\n123' ~ / ^ digit /) {
+  echo 'BUG'
+}
+
+if ($'abc123\n' ~ / digit $ ; reg_newline /) {
+  echo 'yes'
+}
+if ($'abc\n123' ~ / ^ digit ; reg_newline /) {
```

Comment:
BTW as of a few weeks ago, you can use `u'abc\n123'` instead of the `$''` bash syntax.     `u` ensures it's a unicode string, and `b` is any byte string (but they are both of type `Str`, like `$''` is of type `Str`)




Also the eggex spelling of `^` and `$` is `%begin` and `%end` which is a sign it, though we accept the other spellings to be familiar